Stability: Improve Developer Documentation (#8517)
solo-io/gloo: test/e2e/README.md

# In Memory End-to-End tests

> See the [developer e2e testing guide](/devel/testing/e2e-tests.md) for more information on how to run these tests.    

Reviewer: Same suggestion as with the other Readme

However, if you have made changes to the component, refer to the [Envoyinit README](https://github.com/solo-io/gloo/blob/main/projects/envoyinit) for build instructions.

### Run Tests

> The `test` make target runs ginkgo with a set of useful flags. See [run-tests](/devel/testing/run-tests.md) for more details about common techniques for running tests.  The following environment variables can be configured for this target:

Reviewer 2: ```suggestion
The `test` make target runs ginkgo with a set of useful flags. See [run-tests](/devel/testing/run-tests.md) for more details about common techniques and common environment variables used when running tests.  The following environment variables can also be configured for this target:
```

|-------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ENVOY_IMAGE_TAG   | ""      | The tag of the gloo-envoy-wrapper-docker image built during setup                                                                                                                                                                                  |
| TEST_PKG          | ""      | The path to the package of the test suite you want to run                                                                                                                                                                                          |
| WAIT_ON_FAIL      | 0       | Set to 1 to prevent Ginkgo from cleaning up the Gloo Edge installation in case of failure. Useful to exec into inspect resources created by the test. A command to resume the test run (and thus clean up resources) will be logged to the output. |

> | INVALID_TEST_REQS | fail    | The behavior for tests which depend on environment conditions that aren't satisfied. Options are `skip`, `run`, `fail`                                                                                                                             |

Reviewer 2: I don't see a reason to remove this documentation

`TEST_PKG` is still viable and not documented elsewhere
`WAIT_ON_FAIL` and `INVALID_TEST_REQS` are documented below, but I don't see the documentation here as redundant

Author: I'm happy to re-add it. My thought here was that we have some standard environment variables that apply to all tests (not just e2e tests) and I've defined those in the `run-tests.md` docs (this includes TEST_PKG). My hope was that shared variables are in one place, and specific variables for these tests are defined here. If you think we're better off having the duplicate definition of our shared variables, I'm happy to add that

Reviewer 2: Ah yeah, that makes sense, I'll suggest a tweak above instead

#### Use INVALID_TEST_REQS

> Certain test require environmental conditions to be true for them to succeed. For example, certain tests will only run on a Linux machine.

Reviewer 2: ```suggestion
Certain tests require environmental conditions to be true for them to succeed. For example, there are tests that only run on Linux machines.
```
